src/Credentials/AssumeRoleWithWebIdentityCredentialProvider.php

```diff
@@ -0,0 +1,105 @@
+<?php
+namespace Aws\Credentials;
+
+use Aws\Exception\CredentialsException;
+use Aws\Result;
+use Aws\Sts\StsClient;
+use GuzzleHttp\Promise;
+
+/**
+ * Credential provider that provides credentials via assuming a role with a web identity
+ * More Information, see: https://docs.aws.amazon.com/aws-sdk-php/v3/api/api-sts-2011-06-15.html#assumerolewithwebidentity
+ */
+class AssumeRoleWithWebIdentityCredentialProvider
+{
+    const ERROR_MSG = "Missing required 'AssumeRoleWithWebIdentityCredentialProvider' configuration option: ";
+    const ENV_TOKEN_FILE = 'AWS_WEB_IDENTITY_TOKEN_FILE';
+    const ENV_ARN = 'AWS_ROLE_ARN';
+    const ENV_SESSION = 'AWS_ROLE_SESSION';
```

Comment:
This doesn't overlap with the same class constant from `CredentialProvider`, but is it ever used?